clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -78,8 +78,10 @@ invariant_checker_t::report_if_false(per_shard_t *shard, bool condition,
 {
     if (!condition) {
         std::cerr << "Trace invariant failure in T" << shard->tid_ << " at ref # "
-                  << shard->stream->get_record_ordinal() << ": " << invariant_name
-                  << "\n";
+                  << shard->stream->get_record_ordinal() << " ("
+                  << shard->instr_count_since_last_timestamp_
+                  << " instrs since timestamp " << shard->last_timestamp_
+                  << "): " << invariant_name << "\n";
```

Comment:
Let's add a simple test that these new fields are correct: we can easily have a checker unit test verify with a scenario or two of records with timestamps.  Then we'll catch any regression from future code changes right up front in our suite.